semgrep-core/src/core/Rule.ml

```diff
@@ -144,7 +144,7 @@ and metavar_cond =
    * update: this is also useful to keep separate from CondEval for
    * the "regexpizer" optimizer (see Analyze_rule.ml).
    *)
-  | CondRegexp of MV.mvar * regexp
+  | CondRegexp of MV.mvar * regexp * bool (* constant propagation? *)
```

Comment:
It could be nice having a record type for this, although at this point I think a tuple is still fine.